### PR TITLE
Automatic Logger Level Update

### PR DESCRIPTION
**Pull request by williamwen42**
Address #1184 by updating logger levels whenever `torchdynamo.config.log_level` is updated.

Additional changes:
- Cleaned up log level related logic
- Fixed tests that checked for log output
- Revert `log_cli=True` in pytest.ini (#1197) since it pollutes local tests (e.g. `pytest test`).

Torchdynamo test: run a function multiple times and expect different log levels in outputs: `pytest -v -s test/test_misc.py::MiscTests::test_config_log_level`.

PyTorch test: Add `import torchdynamo; import logging; torchdynamo.config.log_level = logging.DEBUG` to a test that purposefully fails (i.e. add `self.assertTrue(False)` to the test `PYTORCH_TEST_WITH_DYNAMO=1 pytest test/test_optim.py -k test_new_pattern_no_warning`)

### CODE REVIEW

File: torchdynamo/config.py [hunk 1]
```diff
@@ -22,12 +22,69 @@
...
+def init_logging():
+    if "PYTEST_CURRENT_TEST" not in os.environ:
+        logging.config.dictConfig(LOGGING_CONFIG)
+        # previous dictConfig call may overwrite logger levels
+        _set_loggers_level(log_level)
+        if log_file_name is not None:
+            log_file = logging.FileHandler(log_file_name)
+            log_file.setLevel(log_level)
+            for logger in _get_loggers():
+                logger.addHandler(log_file)
```
Comment (mlazos) [inline torchdynamo/config.py:82]: Is it possible to leave init_logging and the original dict config inside utils.py and import it here? The main purpose of config.py is to contain config assignments, this feels out of place to me.

Comment (williamwen42) [reply torchdynamo/config.py]: Moved back to utils.

File: torchdynamo/config.py [hunk 2]
```diff
@@ -22,12 +22,69 @@
 # WARN print warnings (including graph breaks)
 # ERROR print exceptions (and what user code was being processed when it occurred)
 log_level = logging.WARNING
-# Verbose will print full stack traces on warnings and errors
-verbose = False
+
+
+def _get_loggers():
```
[the remark this answers is not in the document]
Comment (williamwen42) [reply torchdynamo/config.py]: Moved to utils.

File: torchdynamo/config.py [hunk 3]
```diff
@@ -22,12 +22,69 @@
...
+        logging.getLogger("torchinductor"),
+    ]
+
+
+def _set_loggers_level(level):
+    for logger in _get_loggers():
+        logger.setLevel(level)
+
+
+_set_loggers_level(log_level)
```
[the remark this answers is not in the document]
Comment (williamwen42) [reply torchdynamo/config.py]: Moved to init_logging.

**Review by jansel — APPROVED** [review]

**Review by mlazos — APPROVED** [review]

**Pull request merged**